src/app/view/perspectives/search/all/search-all.component.ts

```diff
@@ -56,18 +64,48 @@ export class SearchAllComponent implements OnInit, OnDestroy {
     this.querySubscription = this.store.select(selectQuery)
       .pipe(
         skipWhile(query => isNullOrUndefined(query)),
-        tap(query => this.store.dispatch(new DocumentsAction.Get({query}))),
         tap(query => this.store.dispatch(new CollectionsAction.Get({query}))),
+        tap(query => this.store.dispatch(new DocumentsAction.Get({query}))),
         tap(query => this.store.dispatch(new ViewsAction.Get({query})))
       ).subscribe();
-    this.documents$ = this.store.select(selectDocumentsByQuery);
-    this.collections$ = this.store.select(selectCollectionsByQuery);
-    this.views$ = this.store.select(selectViewsByQuery);
```

Comment:
Why have you changed the way this was prepared to be used? Isn't it better to be able to get parts of the results instead of waiting for all the responses from the server? The previous implementation also make use of unsubscribing done by the template.

Reply:
The previous implementation didn't allowed me only to use the subscription results only in certain parts of the template. I have `loading()`, `emptySearch()` methods and `*ngIf="collections.length > 0"` conditions for divider lines (so that there are only divider lines between shown perspectives) that need to know the result of all 3 subscriptions.

Reply:
Those cases can be handled differently by combining those observables to a single one or using them at several places. Sure, unless it is more complicated than your current solution. In my opinion, it is better to work just with observables otherwise someone changing your code later might think he works with component state which is not being changed during its lifecycle.

You do not have to change this implementation now but think about it the next time you're either writing a new component or refactoring an existing one. ;)